manuskript/models/abstractModel.py

```diff
@@ -92,9 +92,15 @@ def findItemsContaining(self, text, columns, caseSensitive=False):
         """
         return self.rootItem.findItemsContaining(text, columns, mainWindow(), caseSensitive)
 
-    def getItemByID(self, ID):
+    def getItemByID(self, ID, ignore=None):
+        """Returns the item whose ID is `ID`, unless this item matches `ignore`."""
+
         def search(item):
             if item.ID() == ID:
+                if item == ignore:
+                    # The item we really want won't be found in the children of this
+                    # particular item anymore; stop searching this branch entirely.
+                    return None
```

Comment:
The comment here is misleading, the 'ignore' only ignores if the item with the right ID matches the 'ignore' value, the comment says it skips any items that are children of the item to ignore.

Reply:
It will skip the children of the item currently being compared against. Which makes sense because the IDs are supposed to be unique... except in the exact situation we are dealing with here. The code will only ever loop through the child items when the ID doesn't match in the first place. The new item cannot be a child of the duplicate item to begin with unless we have a bug elsewhere.

The comment is there exactly because the code itself made me second-guess whether the logic was right. (And your later confusion thinking it was a flat loop as opposed to the recursion it is only added to that.)

If this is not what you mean, can you suggest a better comment?

Reply:
Humm.. I hadn't thought of deletion of a folder when we have a folder opened in corkview, I was only thinking of a Text open, and a Text obviously can't have children. +1